packages/headless/src/api/knowledge/baseKnowledgeAPISlice.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  BaseQueryFn,
+  createApi,
+  EndpointDefinition,
+  FetchArgs,
+  fetchBaseQuery,
+  FetchBaseQueryError,
+  retry,
+} from '@reduxjs/toolkit/query';
+import {ConfigurationSection} from '../../state/state-sections';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+/**
+ * `dynamicBaseQuery` is passed to the baseQuery of the createApi,
+ * but note that the baseQuery will not be used if a queryFn is provided in the createApi endpoint
+ */
+const dynamicBaseQuery: BaseQueryFn<
+  string | FetchArgs,
```

Comment:
Because I became confused down the rabbit hole trying to comply with those types and this lost the common sense. Thanks for pointing that out. It had no use.